src/pages/LogDrop/FileDropper.tsx

```diff
@@ -1,63 +1,85 @@
-import { useCallback, useRef, useState } from "react";
+import { useCallback, useTransition } from "react";
 import styled from "@emotion/styled";
 import { palette } from "@leafygreen-ui/palette";
 import { useDropzone } from "react-dropzone";
 import { useLogDropAnalytics } from "analytics";
 import { LogTypes } from "constants/enums";
+import { LOG_FILE_SIZE_LIMIT } from "constants/logs";
 import { size } from "constants/tokens";
 import { useLogContext } from "context/LogContext";
 import { useToastContext } from "context/toast";
-import { leaveBreadcrumb } from "utils/errorReporting";
-import { processLogString } from "utils/string";
+import { leaveBreadcrumb, reportError } from "utils/errorReporting";
+import { fileToStream } from "utils/file";
+import { decodeStream } from "utils/streams";
 import FileSelector from "./FileSelector";
+import LoadingAnimation from "./LoadingAnimation";
 import ParseLogSelect from "./ParseLogSelect";
+import useLogDropState from "./state";
 
 const { green } = palette;
-interface FileDropperProps {
-  onChangeLogType: (logType: LogTypes) => void;
-}
 
-const FileDropper: React.FC<FileDropperProps> = ({ onChangeLogType }) => {
+const FileDropper: React.FC = () => {
   const dispatchToast = useToastContext();
   const { sendEvent } = useLogDropAnalytics();
-  const { ingestLines, setFileName, logMetadata } = useLogContext();
-  const { fileName } = logMetadata ?? {};
-
-  const lineStream = useRef<FileReader["result"]>(null);
-  const [hasDroppedLog, setHasDroppedLog] = useState(false);
+  const { ingestLines, setFileName, setLogMetadata } = useLogContext();
+  const [, startTransition] = useTransition();
+  const { state, dispatch } = useLogDropState();
 
   const onDrop = useCallback(
     (acceptedFiles: File[]) => {
       leaveBreadcrumb("Dropped file", {}, "user");
       sendEvent({ name: "Dropped file" });
-      acceptedFiles.forEach((file) => {
-        const reader = new FileReader();
-
-        reader.onabort = () => dispatchToast.error("File reading was aborted.");
-        reader.onerror = () => dispatchToast.error("File reading failed.");
-        reader.onload = () => {
-          setHasDroppedLog(true);
-          setFileName(file.name);
-          lineStream.current = reader.result;
-        };
-        reader.readAsText(file);
-      });
+      dispatch({ type: "DROPPED_FILE", file: acceptedFiles[0] });
     },
-    [setFileName, dispatchToast, sendEvent]
+    [dispatch, sendEvent]
   );
 
   const onParse = useCallback(
     (logType: LogTypes | undefined) => {
       if (logType) {
-        onChangeLogType(logType);
+        setLogMetadata({ logType });
         leaveBreadcrumb("Parsing file", { logType }, "process");
-        sendEvent({ name: "Processed Log", logType });
-        if (typeof lineStream.current === "string") {
-          ingestLines(processLogString(lineStream.current), logType);
-        }
+        dispatch({
+          type: "PARSE_FILE",
+        });
```

Comment:
nit. (no need to take up 3 lines)
```suggestion
        dispatch({ type: "PARSE_FILE" });
```